### PR TITLE
BiG-CZ Map Popups

### PR DESCRIPTION
**Pull request by arottersman**
## Overview
Adds popovers to the BiG-CZ results we display on the map (Currently CINERGI and WDC). 

**Wireframe**
screen shot 2017-08-22 at 11 18 33 am

Connects #2136 

### Demo

screen shot 2017-08-22 at 11 10 47 am
screen shot 2017-08-22 at 11 11 20 am

th8cajmzhy

### Notes
1. The wireframes don't reflect the actual data for the catalogs, so I just reused the `ResultView` for the popups while we determine what they should contain
2. The interaction for opening a CINERGI popup is weird because you have to click the skinny boundary line. Clicking the inside of the boundaries would also be bad though; boundaries overlap.
3. I didn't match the wireframe "detail" button exactly because setting it to `ui-secondary` seemed much to dark on the white background
3. Refactored the enter-detail-mode triggering to allow the popup access

## Testing Instructions

 * Pull and `bundle`. There are style changes, make sure the sass compiles.
 * Get results for the three tabs by searching something like `water` 
    - Confirm you can still open the details from the sidebar from all three catalogs
    - Confirm if you click on a Cinergi map boundary, its corresponding popup opens and you can use its details button
    - Confirm the same for the WDC points

### CODE REVIEW

File: src/mmw/js/src/data_catalog/models.js [hunk 1]
```diff
@@ -190,13 +199,53 @@ var Results = Backbone.Collection.extend({
...
+            // and those that intersect the area of interest (CINERGI and CUAHSI).
+            filteredResults = _.filter(data.results, function(r) {
+                return r.geom === null || turfIntersect(aoi, r.geom) !== undefined;
+            });
+
+        // Add the result's id onto the geojson's geom so the map can keep
+        // track of which shape came from where
+        return _.map(filteredResults, function(r) {
+            if (r.geom) {
+                r.geom.properties = {
```
Comment (arottersman) [inline src/mmw/js/src/data_catalog/models.js]: We could do this on the backend as well, but if we end up exposing the API, our users might not want the duplicate data clogging up the response.

**Comment by jfrankl** [comment]
When the user clicks on "details" in the popup, the popup should close. Other than that, this is good to go!

FYI, @ajrobbins and I have another concept for the bounding box interaction. We'll write that up in a separate issue.

**Comment by rajadain** [comment]
Waiting for a fixup to come in for the hover issue discussed IRL. Otherwise:

 * I like the popups, and agree that reusing the Result View for this purpose. We should remove the hover interaction of the ResultView in this case.
 * Clicking on the edge of CINERGI results feels a little odd. Good enough for now, but we discussed the possibility of sorting the items on the map to keep the smallest at the top, so you _could_ click on an inside area and it will select the smallest / most locally relevant one. This can be attempted later.
 * When we go into the details view, the map element should be highlighted until the user goes back. I remember there being some discussion of hiding all other map elements, keeping only the selected one, in the details view, but highlighting the selected one seems like an easy thing to do at this stage.

**Comment by arottersman** [comment]
Pushed a commit to modify the highlight behavior.

One quirk is in the details page you can still open additional popups, and they will highlight.

**Comment by rajadain** [comment]
I created #2181 for all the hover issues, even some found herein, so I think this is good to merge. Just going to take a quick peek over the code now.

**Comment by arottersman** [comment]
Thanks, those are good ideas for the hovering. I think it might take some time to get the interactions right, so I'm all for deferring.

File: src/mmw/js/src/data_catalog/models.js (hunk 1 above)
Comment (rajadain) [inline src/mmw/js/src/data_catalog/models.js]: If we ever did #1950 it would assist with this. For now I think this is fine.

**Comment by arottersman** [comment]
Thanks! Going to merge.

**Pull request merged**